misk/src/test/kotlin/misk/hibernate/TruncateTablesService.kt

```diff
@@ -0,0 +1,48 @@
+package misk.hibernate
+
+import com.google.common.util.concurrent.AbstractIdleService
+import com.google.inject.Key
+import misk.DependentService
+import misk.inject.toKey
+import org.hibernate.SessionFactory
+import javax.inject.Provider
+import kotlin.reflect.KClass
+
+/**
+ * Truncate tables before running each test.
+ *
+ * We truncate _before_ tests because that way we always have a clean slate, even if a preceding
+ * test wasn't able to clean up after itself.
+ */
+// TODO(jwilson): promote this to the misk testing library.
```

Comment:
do you think we're going to go down the junit rule route? or promote the idea of services for testing too?

Reply:
Services play better with injection. Also no more rules with JUnit 5!